api/v1/utils/items.js

```diff
@@ -50,9 +50,20 @@ const loadItemKeys = async query => {
 
 const getRecipeFor = async (query, itemname) => {
   try {
-    const statement = `SELECT * FROM synth_recipes AS r
-            JOIN item_basic AS b ON r.result = b.itemid OR resultHQ1 = b.itemid OR resultHQ2 = b.itemid OR resultHQ3 = b.itemid
-            WHERE b.name = ?;`;
+    const statement = `SELECT *,
+    (SELECT name from item_basic WHERE itemid = Ingredient1) AS Ingredient1Name,
+    (SELECT name from item_basic WHERE itemid = Ingredient2) AS Ingredient2Name,
+    (SELECT name from item_basic WHERE itemid = Ingredient3) AS Ingredient3Name,
+    (SELECT name from item_basic WHERE itemid = Ingredient4) AS Ingredient4Name,
+    (SELECT name from item_basic WHERE itemid = Ingredient5) AS Ingredient5Name,
+    (SELECT name from item_basic WHERE itemid = Ingredient6) AS Ingredient6Name,
+    (SELECT name from item_basic WHERE itemid = Ingredient7) AS Ingredient7Name,
+    (SELECT name from item_basic WHERE itemid = Ingredient8) AS Ingredient8Name,
+    (SELECT name from item_basic WHERE itemid = ResultHQ1) AS ResultHQ1Name,
+    (SELECT name from item_basic WHERE itemid = ResultHQ2) AS ResultHQ2Name,
+    (SELECT name from item_basic WHERE itemid = ResultHQ3) AS ResultHQ3Name
```

Comment:
Sounds good! I didn't know it was cached, that makes it a lot easier. I'll implement this along with the suggestion from your other comment.

Reply:
Implemented in latest push as well as the endpoint returning a more structured object. I left in some of the other fields in case they are used but condensed crafts, ingredient, and results for better parsing.